internal/services/compute/shared_image_gallery_data_source.go

```diff
@@ -43,6 +44,13 @@ func dataSourceSharedImageGallery() *pluginsdk.Resource {
 				Computed: true,
 			},
 
+			"image_names": {
+				Type:     pluginsdk.TypeSet,
+				Computed: true,
+				Elem:     &pluginsdk.Schema{Type: pluginsdk.TypeString},
+				Set:      pluginsdk.HashString,
+			},
```

Comment:
This shouldn't need to be a set since it's a data source and every item gets the same hashcode, so this should be a List
```suggestion
			"image_names": {
				Type:     pluginsdk.TypeList,
				Computed: true,
				Elem:     &pluginsdk.Schema{Type: pluginsdk.TypeString},
			},
```